src/Init/Lean/Elab/Term.lean

```diff
@@ -211,7 +223,7 @@ unless ex.data.hasSyntheticSorry $ logMessage ex;
 pure syntheticSorry
 
 def elabTerm (stx : Syntax) (expectedType? : Option Expr) : TermElabM Expr :=
-withNode stx $ fun node => do
+withFreshMacroScope $ withNode stx $ fun node => do
```

Comment:
@leodemoura and `withFreshMacroScope` is used here. We should eventually add it to the other monads too, of course.

Reply:
I still don't see strong evidence for having `withFreshMacroScope` as a method at the `MonadQuotation` class. The generic methods at `Quotation.lean` do not use it. `withFreshMacroScope` can be a regular method of `TermElabM`.

Reply:
That is, users of `MonadQuotation` methods are responsible for creating updating the macroscope. The generic methods only need to query it using `getCurrMacroScope`. Not sure whether you have other plans for it or not

Reply:
The `match_syntax` implementation now has more examples of this function. It is still in `TermElabM`, but should at least show that it is a useful operation to have for code that works with quotations, esp. recursively. Do the new examples make sense to you?
https://github.com/leanprover/lean4/blob/1d2d24f3202f0fd6f40d82f9f7cac61a7ae85fa9/src/Init/Lean/Elab/Quotation.lean#L226-L230